VideoLocker/src/main/java/org/edx/mobile/view/CourseVideoListActivity.java

```diff
@@ -102,6 +126,66 @@ public void onClick(View v) {
     }
 
     @Override
+    public boolean onPrepareOptionsMenu(Menu menu) {
+        PrefManager.UserPrefManager userPrefManager = new PrefManager.UserPrefManager(this);
+
+        if (userPrefManager.isUserPrefVideoModel()) {
+            menu.findItem(R.id.action_change_mode).setIcon(
+                new IconDrawable(this, Iconify.IconValue.fa_film)
+                    .actionBarSize());
+        } else {
+            menu.findItem(R.id.action_change_mode).setIcon(
+                new IconDrawable(this, Iconify.IconValue.fa_list)
+                    .actionBarSize());
+        }
+        return super.onPrepareOptionsMenu(menu);
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        // Handle presses on the action bar items
+        switch (item.getItemId()) {
+            case R.id.action_change_mode:
+                changeMode();
+                return true;
+            default:
+                return super.onOptionsItemSelected(item);
+        }
+    }
+
+    public void changeMode(){
+        //Creating the instance of PopupMenu
+        PopupMenu popup = new PopupMenu(this, this.progressWheel);
+        //Inflating the Popup using xml file
+        popup.getMenuInflater()
+            .inflate(R.menu.change_mode, popup.getMenu());
+        MenuItem menuItem = popup.getMenu().findItem(R.id.change_mode_video_only);
+
+
+        //registering popup with OnMenuItemClickListener
+        popup.setOnMenuItemClickListener(new PopupMenu.OnMenuItemClickListener() {
+            public boolean onMenuItemClick(MenuItem item) {
+                PrefManager.UserPrefManager userPrefManager =
+                    new PrefManager.UserPrefManager(CourseVideoListActivity.this);
+                boolean currentVideoMode = userPrefManager.isUserPrefVideoModel();
+                boolean selectedVideoMode = modeVideoOnly.equals( item.getTitleCondensed() );
```

Comment:
@aleffert  some people proposed this approach
String[] id = getResources().getResourceName(item.getItemId()).split("\\/");

but i think as android does not provide direct API for this, it may not a good idea to hack it?